src/rpc/net.cpp

```diff
@@ -322,6 +323,61 @@ static RPCHelpMan addnode()
     };
 }
 
+static RPCHelpMan addconnection()
+{
+    return RPCHelpMan{"addconnection",
+        "\nOpen an outbound connection to a specified node. This RPC is for testing only.\n",
+        {
+            {"address", RPCArg::Type::STR, RPCArg::Optional::NO, "The IP address and port to attempt connecting to."},
+            {"connection_type", RPCArg::Type::STR, RPCArg::Optional::NO, "Type of connection to open, either \"outbound\" or \"blockrelay\"."},
+        },
+        RPCResult{
+            RPCResult::Type::OBJ, "", "",
+            {
+                { RPCResult::Type::STR, "address", "Address of newly added connection." },
+                { RPCResult::Type::STR, "connection_type", "Type of connection opened." },
```

Comment:
This field confirms what the RPC parsed from the user input per Postel's Law. If you pass " BLOCKRELAY   ", it's essential to know that it was correctly understood as "blockrelay".

Reply:
If the args can't be parsed, a parse failure will be thrown already https://github.com/bitcoin/bitcoin/pull/19315/files#r533936622

Reply:
Yes, but in the absence of failure, feedback is better than uncertainty or requiring the user to check the source code.

Reply:
I feel completely indifferent as to whether there's a return result or not. I believe the initial implementation left it out but then I got review comments requesting it be added. 

Happy to do whatever reviewers prefer. To me it doesn't feel super important one way or the other since its a test-only RPC so we don't have to adhere to normal RPC deprecation rules.

Reply:
@jonatack I don't have a strong opinion on this, I was more surprised to notify the discrepancy with other usual RPCs. Maybe this RPC design principle should be documented in https://github.com/bitcoin/bitcoin/blob/master/doc/JSON-RPC-interface.md ? The Postel's Law sounds a rational good enough to me but I'm not heavily developing against Core RPC so don't feel competent here to evaluate here :)

Reply:
It seems no-one has a strong opinion on this. I think it's fine as is. Resolve?

Reply:
That paper on Postel's Law appears to discuss issues of computational complexity involving difficult parsing or high ambiguity, which IIRC isn't the case for this RPC, and states among other things that a balance should be struck, which I agree with. In security-sensitive missions for airports, power grids, industry manufacturers, etc., my experience was that it depended on the context, stakeholders and product managers. Usually, anyone whose code didn't handle simple, unambiguous input in a robust and non-annoying way got a bug report and a phone call by a PM or scrum master, so it became an instinct to do so. I'm unlearning here to adapt to the consensus of the stakeholders on this project. It's a process!